src/utils/notification.utils.ts

```diff
@@ -0,0 +1,147 @@
+import mongoose from 'mongoose';
+import {
+  ExpoNotification,
+  MongoNotification,
+  NotificationData,
+} from '../interfaces/Notification.interface';
+import Notification from '../models/Notification.model';
+import User from '../models/User.model';
+
+/**
+ * Sends notification to specified user(s) through expo and saves notification to database.
+ * If userId is a string, sends notification to a single user. If it's an array of strings,
+ * sends notification to all users in the array.
+ *
+ * If the userIds field is an empty array, the function will return an empty array.
+ * If the userId is invalid, the function will return undefined. If a userId in the array is invalid, the function
+ * will ignore that userId and continue to send notifications to the other valid userIds.
+ *
+ * @param {string[]} userIds - The id or ids of the user(s) to receive the notification.
+ * @param {string} title - The title of the notification.
+ * @param {string} body - The content of the notification.
+ * @param {number} delay - Optional delay until the notification should be shown to the user. (set as 0 for now)
+ * @param {NotificationData} data - Optional data to include with the notification.
+ *
+ * @return {Promise<Notification[]>} - Returns a promise that resolves to an array of notifications.
+ */
+export const sendNotifications = async (
+  userIds: string[],
+  title: string,
+  body: string,
+  data: NotificationData,
+  isPush: boolean,
+  delay: number = 0
+) => {
+  // array of notifications to return
+  let notifications: any[] = [];
```

Comment:
hello sir, may i know what language you are programming in? i think it rhymes with "hype"